File: eggs/Ice_Cream_Sandwich/src/main/java/com/android_i/egg/PlatLogoSnapshotProvider.java
package com.android_i.egg;

import android.content.Context;
import android.view.View;
import android.widget.ImageView;


public class PlatLogoSnapshotProvider extends com.dede.basic.PlatLogoSnapshotProvider {
    @Override
    public View create(Context context) {
        ImageView mContent = new ImageView(context);
        mContent.setImageResource(R.drawable.i_platlogo);
        mContent.setScaleType(ImageView.ScaleType.CENTER_INSIDE);
        return mContent;
    }
}

File: eggs/U/src/main/java/com/android_u/egg/PlatLogoSnapshotProvider.kt
package com.android_u.egg

import android.content.Context
import android.view.Gravity
import android.view.View
import android.widget.FrameLayout
import android.widget.ImageView
import com.android_u.egg.PlatLogoActivity.Starfield
import com.dede.basic.PlatLogoSnapshotProvider
import java.util.Random

class PlatLogoSnapshotProvider : PlatLogoSnapshotProvider() {

    override fun create(context: Context): View {
        val dp = context.resources.displayMetrics.density
        val random = Random()
        val starfield = Starfield(random, dp * 2.0f)
        starfield.setVelocity(
            (random.nextFloat() - 0.5f) * 200.0f,
            (random.nextFloat() - 0.5f) * 200.0f
        )
        val layout = FrameLayout(context)
        layout.background = starfield

        val logo = ImageView(context)
        logo.setImageResource(R.drawable.u_platlogo)
        val lp: FrameLayout.LayoutParams = FrameLayout.LayoutParams(
            FrameLayout.LayoutParams.MATCH_PARENT,
            FrameLayout.LayoutParams.MATCH_PARENT
        )
        lp.gravity = Gravity.CENTER
        layout.addView(logo, lp)

        return layout
    }
}